sdk/core/core-amqp/test/requestResponse.spec.ts

```diff
@@ -399,4 +409,117 @@ describe("RequestResponseLink", function() {
       );
     }
   });
+
+  describe("sendRequest clears timeout", () => {
+    const _global = getGlobal();
+    const originalClearTimeout = clearTimeout;
+    afterEach(() => {
+      _global.clearTimeout = originalClearTimeout;
```

Comment:
Oh no....

Reply:
Unfortunately I don't know a better way to test this. The test will exit pass without clearing the timeout, but the process will stay open another minute.

Reply:
In another set of tests I just pass setTimeout and clearTimeout into the method as parameters.

Reply:
You know, another thing is that I think sinon let's you control the clock. So you could just fast-forward and ensure your timer doesn't fire.

Reply:
Since this is part of the public API I don't _really_ want to change `sendRequest` to accept those.

Reply:
I'm not sure if controlling the clock would help. There's not a way for me to get a reference to the timer from outside of `sendRequest`, unless I spy on setTimeout and clearTimeout :-/ And I also don't have access to the method that the timeout invokes :-(

Reply:
All good points. Appreciate you thinking about it.